.github/workflows/build-and-publish.yaml

```diff
@@ -0,0 +1,157 @@
+    runs-on: 'ubuntu-latest'
+    outputs:
+      digest: ${{ steps.set-output.outputs.digest }}
+    steps:
+      - name: Set Output
+        id: set-output
+        run: echo "::set-output name=digest::centos/redis-32-centos7@sha256:06dbb609484330ec6be6090109f1fa16e936afcf975d1cbc5fff3e6c7cae7542"
+
+  operator-image:
+    name: Publish 3.5 Operator Image
+    runs-on: 'ubuntu-latest'
+    env:
+      OPERATOR_TAG: quay.io/projectquay/quay-operator:3.5
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+        with:
+          ref: 'redhat-3.5'
+
+      - name: Login to Quay.io
+        env:
+          QUAY_USER: projectquay+quay_github
+        run: |
+          echo "::add-mask::${{ secrets.QUAY_TOKEN }}"
+          docker login -u "${QUAY_USER}" -p '${{ secrets.QUAY_TOKEN }}' quay.io
+
+      - name: Build Operator
+        run: docker build -t "${OPERATOR_TAG}" .
+
+      - name: Push Operator Image
+        run: docker push "${OPERATOR_TAG}"
+
+  operator-index-images:
+    name: Publish 3.5 Catalog Index Image
+    runs-on: 'ubuntu-latest'
+    needs: [quay-image, clair-image, builder-image, qemu-builder-image, postgres-image, redis-image]
+    env:
+      INDEX_TAG: quay.io/projectquay/quay-operator-index:3.5
+      Y_STREAM: 3.5
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+        with:
+          ref: 'redhat-3.5'
+
+      - name: Login to Quay.io
+        env:
+          QUAY_USER: projectquay+quay_github
+        run: |
+          echo "::add-mask::${{ secrets.QUAY_TOKEN }}"
+          docker login -u "${QUAY_USER}" -p '${{ secrets.QUAY_TOKEN }}' quay.io
+
+      - name: Update CSV Image Versions
+        uses: mikefarah/yq@master
+        with:
+          cmd: |
+            yq eval -i '
+              .metadata.annotations.quay-version = "env(Y_STREAM)" |
+              .metadata.annotations.containerImage = "quay.io/projectquay/quay-operator:env(Y_STREAM)" |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].image = "quay.io/projectquay/quay-operator:env(Y_STREAM)" |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].env[4].value = "${{needs.quay-image.outputs.digest}}" |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].env[5].value = "${{needs.clair-image.outputs.digest}}" |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].env[6].value = "${{needs.builder-image.outputs.digest}}" |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].env[7].value = "${{needs.qemu-builder-image.outputs.digest}}" |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].env[8].value = "${{needs.postgres-image.outputs.digest}}" |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].env[9].value = "${{needs.redis-image.outputs.digest}}"
```

Comment:
These lines are fragile (`env[4-9]`). Using annotations (i.e. `.metadata.annotations.quay-version = "env(Y_STREAM)"`) is a better idea, but since it requires changes to the CSV yaml I'll do that in a separate PR.

Reply:
Assuming that we would like to use the "last 3.5 build" of quay, clair and builder why not replace only the image tag to point to `v3.5`? i.e. we would not need to know what is the current sha. wdyt?

Reply:
That's what I tried first, I moved to using the sha because the operator refused to deploy quay via tag only. Not sure if this could behavior be disabled though.

Reply:
Yes, you are right: https://github.com/quay/quay-operator/blob/redhat-3.5/pkg/kustomize/kustomize.go#L73-L75

Reply:
Ah thanks for finding the source!